File: Morgan and a String.py
T = int(input())
for x in range(T):
	A = input()
	B = input()
	index_A = 0
	index_B = 0
	final = ""
	len_A = len(A)
	len_B = len(B)
	A += 'a'
	B += 'b'
	for p in range(0,len_A + len_B):
		if ord(A[index_A]) < ord(B[index_B]):
			final += A[index_A]
			index_A += 1
		elif ord(A[index_A]) > ord(B[index_B]):
			final += B[index_B]
			index_B += 1
		else:
			n = 0
			while(ord(A[index_A]) == ord(B[index_B])):
				index_A += 1
				index_B += 1
				n += 1
			if(ord(A[index_A]) < ord(B[index_B])):
				index_A -= n
				index_B -= n
				final += A[index_A]
				index_A += 1
			elif(ord(A[index_A]) > ord(B[index_B])):
				index_A -= n
				index_B -= n
				final += B[index_B]
				index_B += 1
	print(final)

File: Morgan and a String.cpp
#include <iostream>
#include <string>
using namespace std;

int main() 
{
	//int T,i,lo,index_A = 0,index_B = 0,len_A,len_B;
	//string A,B,small;
	//char final[200010];
	int T,i;
	cin>>T;
	for(i=0;i<T;i++)
	{
		int T,i,lo,index_A = 0,index_B = 0,len_A,len_B,n;
		string A,B,small;
		char final[200010];
		cin>>A>>B;
		len_A = A.length();
		len_B = B.length();
		A.append("a");
		B.append("b");
		//lo = len_A < len_B ? len_A : len_B;
		//small = len_A < len_B ? A : B;
		for(i = 0 ; i < len_A + len_B ; i++)
		{
			if((int)A[index_A] < (int)B[index_B])
			{
				final[i] = A[index_A];
				index_A ++;
			}
			else if((int)A[index_A] > (int)B[index_B])
			{
				final[i] = B[index_B];
				index_B ++;
			}
			else
			{
				n = 0;
				while((int)A[index_A] == (int)B[index_B])
				{
					index_A ++;
					index_B ++;
					n++;
				}
				if((int)A[index_A] < (int)B[index_B])
				{
					index_A -= n;
					index_B -= n;
					final[i] = A[index_A];
					index_A ++;
				}
				else if((int)A[index_A] > (int)B[index_B])
				{
					index_A -= n;
					index_B -= n;
					final[i] = B[index_B];
					index_B ++;
				}
			}
		}
		for(i = 0 ; i < len_A + len_B ; i++)
		{
			cout<<final[i];
		}
		cout<<endl;
	}
	return 0;
}